Src/VsVimShared/VsVimHost.cs

```diff
@@ -64,76 +66,165 @@ internal static void Initialize(IVimGlobalSettings globalSettings, IVimApplicati
                 globalSettings.AddCustomSetting(UseEditorDefaultsName, UseEditorDefaultsName, settingsSource);
                 globalSettings.AddCustomSetting(UseEditorTabAndBackspaceName, UseEditorTabAndBackspaceName, settingsSource);
                 globalSettings.AddCustomSetting(UseEditorCommandMarginName, UseEditorCommandMarginName, settingsSource);
-                globalSettings.AddCustomSetting(CleanMacros, CleanMacros, settingsSource);
+                globalSettings.AddCustomSetting(CleanMacrosName, CleanMacrosName, settingsSource);
+                globalSettings.AddCustomSetting(HideMarksName, HideMarksName, settingsSource);
             }
 
             SettingValue IVimCustomSettingSource.GetDefaultSettingValue(string name)
             {
-                return SettingValue.NewToggle(true);
+                switch (name)
+                {
+                    case UseEditorIndentName:
+                    case UseEditorDefaultsName:
+                    case UseEditorTabAndBackspaceName:
+                    case UseEditorCommandMarginName:
+                    case CleanMacrosName:
+                        return SettingValue.NewToggle(false);
+                    case HideMarksName:
+                        return SettingValue.NewString("");
+                    default:
+                        Debug.Assert(false);
+                        return SettingValue.NewToggle(false);
+               }

```

Comment:
Yeah, one of the areas I was getting bogged down in was how best to handle a new non-toggle custom setting.